### DISCUSSION TITLE
restyling

### DISCUSSION THREAD

**Discussion by Marcin-Borowski** [carbon-design-system/carbon-components-svelte#1721] [Q&A] [answered]
Do you have a stack to restyle the component library? I mean to prepare CSS styles (i.e. in a separate subproject) with changed colours.  Adding an SCSS file to the project makes the compilation process very time-consuming, even a couple of minutes. It is tough to work with the HMR approach.
So my idea is to prepare the changed CSS file and attach it as a CSS file - in this case, it works quite well.

**Comment by theetrain** [comment, marked as answer]
Hello, at the moment it's only possible to theme the library with CSS variables or with the [Theme component](https://carbon-components-svelte.onrender.com/components/Theme#custom-theme). Once we adopt Carbon v11 styles, it'll be easier to provide overrides to CSS variables in the Theme component or in plain CSS such as:
```css
:root {
  /* variables here */
}
```

If you're asking about providing full replacement styles, that's something I've looked into, but there are some technical constraints such as controlling how styles are compiled. I've opened a ticket here that may interest you: https://github.com/sveltejs/svelte/issues/8354
